liquibase-maven-plugin/src/main/java/org/liquibase/maven/plugins/AbstractLiquibaseMojo.java

```diff
@@ -839,19 +841,31 @@ public void configureFieldsAndValues() throws MojoExecutionException, MojoFailur
         if (propertyFile != null) {
             getLog().info("Parsing Liquibase Properties File");
             getLog().info("  File: " + propertyFile);
-            try (InputStream is = handlePropertyFileInputStream(propertyFile)) {
+            try (InputStream _is = handlePropertyFileInputStream(propertyFile)) {
+                InputStream is = _is;
                 if (is == null) {
-                    throw new MojoExecutionException(FileUtil.getFileNotFoundMessage(propertyFile));
+                    if (ChangeLogParserConfiguration.WARN_ON_MISSING_CHANGELOGS.getCurrentValue()) {
+                        Scope.getCurrentScope().getLog(getClass()).warning(FileUtil.getFileNotFoundMessage(propertyFile));
+                        is = new ByteArrayInputStream(FileUtil.EMPTY_FILE.getBytes(StandardCharsets.UTF_8));
+                    } else {
+                        throw new MojoExecutionException(FileUtil.getFileNotFoundMessage(propertyFile));
+                    }
                 }
 
                 parsePropertiesFile(is);
                 getLog().info(MavenUtils.LOG_SEPARATOR);
             } catch (IOException e) {
                 throw new UnexpectedLiquibaseException(e);
             }
-            try (InputStream is = handlePropertyFileInputStream(propertyFile)) {
+            try (InputStream _is = handlePropertyFileInputStream(propertyFile)) {
```

Comment:
I would suggest here moving below lines into the above try and remove this piece of code as except these lines it looks like the same:
```
LiquibaseConfiguration liquibaseConfiguration = Scope.getCurrentScope().getSingleton(LiquibaseConfiguration.class);
                final DefaultsFileValueProvider fileProvider = new DefaultsFileValueProvider(is, "Property file "+propertyFile);
                liquibaseConfiguration.registerProvider(fileProvider);
```